### PR TITLE
ROB: Some attributes not copied in DictionaryObject._clone

### LINKED ISSUE #1615: Output pdf has wrong colour, incorrect translation of markup, and wrong scaling
I am converting a PDF to A4 paper size and the output pdf has several bugs:

1. Wrong colour of markup. Gone from Red to black.
2. Incorrect translation of markup. It is slightly to the right whereas it should be exactly overlapping the background grid.
3. Wrong scaling (see comment in the attached screenshot)
4. Lack of central alignment as per transformation in code. Expectation is that there should be equal gap on top and bottom of the page, but the output has only empty space on top.
5. The thick and thin lines of the background grid have been interchanged (strangest bug).

The input file is:
[graph_letter.pdf](https://github.com/py-pdf/pypdf/files/10671366/graph_letter.pdf)

The output file is:
[graph_letter_output.pdf](https://github.com/py-pdf/pypdf/files/10671367/graph_letter_output.pdf)

A side by side comparison is:
comparison

I am using pypdf-3.4.0 in Google Colab.
The code used is:
```python
from pypdf import PdfReader, PdfWriter, Transformation, PageObject, PaperSize
from pypdf.generic import RectangleObject

reader = PdfReader("input.pdf")
page = reader.pages[0]
writer = PdfWriter()

A4_w = PaperSize.A4.width
A4_h = PaperSize.A4.height

# resize page to fit *inside* A4
h = float(page.mediabox.height)
w = float(page.mediabox.width)
scale_factor = min(A4_h / h, A4_w / w)
print(scale_factor)

transform = (
    Transformation()
    .scale(scale_factor, scale_factor)
    .translate(A4_w / 2 - w * scale_factor / 2, A4_h / 2 - h * scale_factor / 2)
)
page.add_transformation(transform)

page.cropbox = RectangleObject((0, 0, A4_w, A4_h))

# merge the pages to fit inside A4

# prepare A4 blank page
page_A4 = PageObject.create_blank_page(width=A4_w, height=A4_h)
page.mediabox = page_A4.mediabox
page_A4.merge_page(page)

writer.add_page(page_A4)
writer.write("output.pdf")
```

Please let me know if I am doing something wrong.

### LINKED ISSUE #1673: writer.clone_document_from_reader(reader) issue
When trying to clone certain pdf documents, an "AttributeError: 'TextStringObject' object has no attribute '_clone'" appears. This problem happens consistently with the same "AttributeError: 'TextStringObject'. 

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
# Windows-10-10.0.19045-SP0

$ python -c "import pypdf;print(pypdf.__version__)"
# 3.4.1
```

## Code + PDF

```python
# TODO: Your code goes here
`   
    # Read the file
    reader = PdfReader(in_file_path, strict = False) # strict = False
    file_name = os.path.basename(in_file_path) # get file name of pdf
    
    # Open a writer file and clone
    writer = PdfWriter()
    writer.clone_document_from_reader(reader) # FAILS IN THIS METHOD
`
```

A few examples of the PDF documents that fail when trying to clone.

[budgeting-loan-form-sf500.pdf](https://github.com/py-pdf/pypdf/files/10848750/budgeting-loan-form-sf500.pdf)
[industrial-injuries-disablement-claim-form-bi100a_1.pdf](https://github.com/py-pdf/pypdf/files/10848751/industrial-injuries-disablement-claim-form-bi100a_1.pdf)

## Traceback

This is the complete Traceback I see:

pypdf\generic\_data_structures.py:233, in DictionaryObject._clone(self, src, pdf_dest, force_duplicate, ignore_fields)
    231                         cur_obj = None
    232                 for (s, c) in objs:
--> 233                     c._clone(s, pdf_dest, force_duplicate, ignore_fields + [k])
    235 for k, v in src.items():
    236     if k not in ignore_fields:

AttributeError: 'TextStringObject' object has no attribute '_clone'

### PR DESCRIPTION
**Pull request by pubpub-zz**
fixes  #1615, #1671

"/N" attributes wrongly ignored during copy process
updated to fix #1673 : 
the pointed object needs to be a dictionary

### CODE REVIEW

File: pypdf/generic/_data_structures.py [hunk 1]
```diff
@@ -194,11 +194,27 @@ def _clone(
         """
         #  First check if this is a chain list, we need to loop to prevent recur
         if (
-            ("/Next" not in ignore_fields and "/Next" in src)
-            or ("/Prev" not in ignore_fields and "/Prev" in src)
+            (
```
Comment (MartinThoma) [inline pypdf/generic/_data_structures.py:197]: A suggestion to make this whole condition a bit shorter:

```
if any(field not in ignore_fields and field in src and isinstance(src.raw_get(field), IndirectObject) for field in ["/Next", "/Prev", "/N", "/V"])
```

Comment (MartinThoma) [reply pypdf/generic/_data_structures.py:197]: And maybe we can give this condition a name, e.g.

```
is_chain_list = any(field ...)
if is_chain_list:
    ...
```

**Comment by MartinThoma** [comment]
@pubpub-zz  Is this a robustness issue or a bug we had in pypdf? So was the issue happening for non-compliant PDFs only?

**Comment by pubpub-zz** [comment]
Half and half
It is a bug that makes some pdf not completely copied

**Comment by pubpub-zz** [comment]
@MartinThoma you need more info?

**Comment by pubpub-zz** [comment]
stdby on this one I may need to complete it

**Comment by pubpub-zz** [comment]
sounds good now

**Pull request merged**